search/includes/classes/class-queue.php

```diff
@@ -18,6 +18,13 @@ class Queue {
 
 	public $schema;
 
+	public const INDEX_COUNT_CACHE_GROUP = 'vip_search';
+	public const INDEX_COUNT_CACHE_KEY = 'index_op_count';
+	public const INDEX_QUEUEING_ENABLED_KEY = 'index_queueing_enabled';
+	public static $max_indexing_op_count = 3000 + 1; // 10 requests per second plus one for clealiness of comparing with Search::index_count_incr
+	private const INDEX_COUNT_TTL = 5 * MINUTE_IN_SECONDS; // Period for indexing operations
+	private const INDEX_QUEUEING_TTL = 1 * HOUR_IN_SECONDS; // Keep indexing op queueing for an hour once ratelimiting is triggered
```

Comment:
This cool-down period seems a little long...something like 15m would still even things out while allowing a faster return to normalcy.